src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.h

```diff
@@ -77,6 +77,11 @@ struct grpc_tls_certificate_provider
   // instances of that provider implementation.
   virtual grpc_core::UniqueTypeName type() const = 0;
 
+  // Returns an OK status if the credentials held by the provider are valid.
+  // What it means for a credential to be valid is determined by the provider
+  // implementation.
+  virtual absl::Status ValidateCredentials() const { return absl::OkStatus(); }
```

Comment:
As mentioned elsewhere, I don't think this method belongs on the base class, only on the file-watcher and static implementations.

Reply:
Removed from the base class.